### PR TITLE
Experiments Creation Flow

### PR DESCRIPTION
**Pull request by liyiy**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

Preliminary create experiment carousel flow, pending https://github.com/PostHog/posthog/pull/7492 and figma designs  

https://user-images.githubusercontent.com/25164963/144765340-1932ebeb-6e77-4810-a96f-67734191f234.mov

## How did you test this code?

*Please describe.*

### CODE REVIEW

File: frontend/src/scenes/experiments/Experiment.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                        <PageHeader title={experiment?.name} />
+                        <div>{experiment?.description}</div>
+                        <div>Owner: {user?.first_name}</div>
+                        <div>Feature flag key: {experiment?.feature_flag}</div>
+                        <Row>
+                            <Col>
+                                <Row>Person allocation</Row>
+                                <Row>The following users will participate in the experiment</Row>
+                                <ul>
+                                    {experiment?.filters?.map((filter: PersonPropertyFilter, idx: number) => (
```
Comment (neilkakkar) [inline frontend/src/scenes/experiments/Experiment.tsx:97]: Note that: The filters we choose in the "Persons to select" are actually the `properties` attribute of `experiment.filters` .

The rest of the filterType is built by the "funnel selecting" page in the middle, which i guess is missing right now. (We don't want to allow selecting filters there - apply em implicitly, no breakdowns either, just the funnel steps)

If nothing is selected here, it means it applies to All users.

**Comment by neilkakkar** [comment]
Starting to come together, excellent! :)

File: frontend/src/scenes/experiments/Experiment.tsx (hunk 1 above)
Comment (liyiy) [reply frontend/src/scenes/experiments/Experiment.tsx:97]: do you mean like filters = { properties: [] } ?

**Comment by liyiy** [comment]
Let me know if there's anything in particular you'd like to see from this PR @neilkakkar, otherwise I would also defer to merging this so I can keep building on top of it :)

File: frontend/src/scenes/experiments/Experiment.tsx (hunk 1 above)
Comment (neilkakkar) [reply frontend/src/scenes/experiments/Experiment.tsx:97]: yes, precisely. (So effectively, the form element is populating the properties of filters for experiment, vs filters itself)

**Comment by neilkakkar** [comment]
Just this one thing (and the related failing static analysis)^, apart from that, happy to go merge & build on top :)

File: frontend/src/types.ts [hunk 2]
```diff
@@ -1310,7 +1310,7 @@ export interface Experiment {
     name: string
     description: string
     feature_flag: string[]
-    filters: Partial<FilterType>
+    filters: Record<string, any>
```
Comment (neilkakkar) [inline frontend/src/types.ts:1313]: `filters` being a `FilterType` is indeed correct, isn't it?

File: frontend/src/scenes/experiments/Experiment.tsx [hunk 3]
```diff
@@ -84,32 +84,34 @@ export function Experiment(): JSX.Element {
...
+                            <PageHeader title={experiment.name} />
+                            <div>{experiment?.description}</div>
+                            <div>Owner: {user?.first_name}</div>
+                            <div>Feature flag key: {experiment?.feature_flag}</div>
+                            <Row>
+                                <Col>
+                                    <Row>Person allocation</Row>
+                                    <Row>The following users will participate in the experiment</Row>
+                                    <ul>
+                                        {experiment.filters?.map((filter: PersonPropertyFilter, idx: number) => (
```
Comment (neilkakkar) [inline frontend/src/scenes/experiments/Experiment.tsx:98]: For the typing fix, this should probably instead be this?
```suggestion
                                        {experiment.filters?.properties.map((filter: PersonPropertyFilter, idx: number) => (
```

File: frontend/src/types.ts (hunk 2 above)
Comment (liyiy) [reply frontend/src/types.ts:1313]: yeah but typescript hates it 😭 and I noticed we do that with a bunch of other keys in our interfaces that should really also be `FilterType`s

**Review by neilkakkar — APPROVED** [review]

**Pull request merged**